GitExtUtils/ArgumentBuilder.cs

```diff
@@ -77,6 +76,12 @@ public void AddRange(IEnumerable<string> args)
             }
         }
 
+        /// <summary>
+        /// Gets the length of the expected command.
+        /// </summary>
+        /// <returns>THe length of the expected command.</returns>
+        public virtual int GetLength() => _arguments.Length;
```

Comment:
This doesn't need to be converted to a method to be made virtual. Personally I prefer it as a property.

Reply:
Yes, correct, it doesn't.
The reason I changed it to a method because an override may performs calculations and, thus, a result would be non-trivial/non-deterministic.
A property is typically safe to reuse across multiple call sites, however you won't invoke a method over and over again.

Reply:
nitpick: commit message protected instead of public

Reply:
Thank you, forgot to amend the message

Reply:
> an override may performs calculations and, thus, a result would be non-trivial/non-deterministic

Does the existing override do that? There are lots of virtual properties in the BCL for which overrides may do whatever they like, but the API reads better with them as a property. It does not feel like querying the length of something should be particularly expensive, and certainly should be idempotent.

I think years of Java gave me a distaste for 'get' methods :)

Reply:
Fair enough, I changed it back to a property.
I have a dislike for Java syntax too :)